src/uint.rs

```diff
@@ -149,16 +150,17 @@ impl Unsigned for UTerm {
 /// ```
 #[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Copy, Hash, Debug, Default)]
 pub struct UInt<U, B> {
-    _marker: PhantomData<(U, B)>,
+    /// The more significant bits of `Self`.
+    pub(crate) msb: U,
+    /// The least significant bit of `Self`.
+    pub(crate) lsb: B,
 }
```

Comment:
As before, we modify `UInt` such that `U` and `B` exist in the value domain. This allows us to remove uses of unsafe that were needed to construct values of `U` and `B` out of thin air.